services/blocks/standard/handler.go

```diff
@@ -743,17 +743,24 @@ func (s *Service) beaconCommittee(ctx context.Context,
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to fetch beacon committees")
 	}
+	log.Debug().Uint64("slot", uint64(slot)).Msg("Obtained beacon committees from API")
+
 	for _, chainBeaconCommittee := range chainBeaconCommittees {
-		if chainBeaconCommittee.Slot == slot && chainBeaconCommittee.Index == index {
-			beaconCommittee = &chaindb.BeaconCommittee{
-				Slot:      slot,
-				Index:     index,
-				Committee: chainBeaconCommittee.Validators,
-			}
-			beaconCommittees[slot][index] = beaconCommittee
-			log.Debug().Uint64("slot", uint64(slot)).Uint64("index", uint64(index)).Msg("Obtained beacon committee from API")
-			return beaconCommittee, nil
+		newBeaconCommittee := &chaindb.BeaconCommittee{
+			Slot:      chainBeaconCommittee.Slot,
+			Index:     chainBeaconCommittee.Index,
+			Committee: chainBeaconCommittee.Validators,
 		}
+		// beaconCommittees cache is slot-scoped, no point adding other slots
```

Comment:
beaconCommittees is shared for all attestations in the block, which could come from different slots, so I think this restriction could result in multiple fetches for a block when only one would be required.  I don't see any harm in removing the conditional here, and storing all committees in the map.

Reply:
This is how it was originally implemented but then I then I added a log line to see the cache hits in action. I realised that this `map` is instantiated in [updateAttestationsForBlock](https://github.com/wealdtech/chaind/blob/master/services/blocks/standard/handler.go#L237) for every blockRoot/slot pair and thus not shared between blocks.
Ideally we would want to fetch once per epoch and cache it for all slots, but that's a bit more involved.